aws/resource_aws_datasync_location_smb.go

```diff
@@ -0,0 +1,220 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/datasync"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsDataSyncLocationSmb() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDataSyncLocationSmbCreate,
+		Read:   resourceAwsDataSyncLocationSmbRead,
+		Update: resourceAwsDataSyncLocationSmbUpdate,
+		Delete: resourceAwsDataSyncLocationSmbDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"agent_arns": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"domain": {
```

Comment:
Since the `Update` function does not support in-place updates of this argument, it should be marked `ForceNew: true` 👍